### ISSUE TITLE
torch.sum() for sparse tensor

### ISSUE THREAD

**Issue by weiyangfb** [pytorch/pytorch#12241]
- Add `torch.sparse_sum(input, dim, keepdim=False, dtype=None) → SparseTensor`
- Add `torch.sparse_sum(input, dtype=None) → Tensor`
- Add autograd support

**Comment by SsnL** [comment]
`sparse_sum` w/ autograd is implement-able with `t.values().sum()` after #11253 is merged.... sorry, I will merge it after finishing this patch I'm working on.

**Comment by weiyangfb** [comment]
@SsnL Awesome! I can assume `sparse_sum()` over dense dims already have autograd support for now :)

**Comment by SsnL** [comment]
yep, summing over sparse dims still needs some more work.

**Comment by vadimkantorov** [comment]
A minor naming question: what is the motivation for prefixing it `sparse_sum` (contrary to a polymorphic `sum`)? Thanks!
